### PR TITLE
Remove all dependency on github secret for pull request 

### PR DESCRIPTION
**Pull request by evalsocket**
## Read then delete

# TL;DR
1. Remove github secret dependency from pull request  

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

### CODE REVIEW

**Comment by evalsocket** [comment]
@EngHabu check workflow testing result - https://github.com/evalsocket/flytepropeller/pull/3/checks?check_run_id=952296324

**Comment by kumare3** [comment]
@evalsocket this is really cool, thank you so much, so in flyte we manage all the CI centrally using https://github.com/lyft/boilerplate. We basically modify boilerplate and sync all repos, so that all repos automatically get all the improvements.
What you have done is awesome.

@EngHabu should we directly add @evalsocket's work to boilerplate? Or should we let him merge here and then migrate the work to boilerplate.

LGTM!

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -26,29 +26,13 @@ jobs:
...
-        with:
-          username: "${{ github.actor }}"
-          password: "${{ secrets.GITHUB_TOKEN }}"
-          image_name: ${{ secrets.flytegithub_repo }}/flytepropeller
-          image_tag: latest,${{ github.sha }},${{ needs.bump-version.outputs.version }}
-          push_git_tag: true
-          registry: docker.pkg.github.com
-          build_extra_args: "--compress=true"
-      - uses: engineerd/setup-kind@v0.4.0
+        name: "Checkout"
```
Comment (EngHabu) [inline .github/workflows/master.yml:29]: Let's keep master.yml as is... this is still needed for master merges.

Comment (EngHabu) [reply .github/workflows/master.yml:29]: You can do [this trick](https://github.community/t/stop-github-actions-running-on-a-fork/17965/2) to disable running it on master merges on forks

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -8,39 +8,13 @@ jobs:
...
-      - name: Push Docker Image to Dockerhub
-        uses: whoan/docker-build-with-cache-action@v5
-        with:
-          username: "${{ secrets.DOCKERHUB_USERNAME }}"
-          password: "${{ secrets.DOCKERHUB_PASSWORD }}"
-          image_name: ${{ secrets.DOCKERHUB_IMAGE_NAME }}
-          image_tag: ${{ github.sha }}
-          push_git_tag: true
-          build_extra_args: "--compress=true"
+          docker build -t lyft/flytepropeller/flytepropeller:${{ github.sha }} .
```
Comment (EngHabu) [inline .github/workflows/pull_request.yml:15]: Can we continue to use the kind action step? I do like the separation in actions to make it easier to debug failures when they happen...

Comment (evalsocket) [reply .github/workflows/pull_request.yml:15]: ok

File: .github/workflows/master.yml (hunk 1 above)
Comment (evalsocket) [reply .github/workflows/master.yml:29]: ok

**Comment by evalsocket** [comment]
@EngHabu all changes are done. I also removed the flytepropeller_repo from master

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -88,4 +86,4 @@ jobs:
         env:
           GO111MODULE: "on"
         with:
-          args: make install && make lint
+          args: make install && make lint
```
Comment (EngHabu) [inline .github/workflows/master.yml:89]: nit
```suggestion
          args: make install && make lint

```

File: .github/workflows/master.yml [hunk 4]
```diff
@@ -40,15 +40,13 @@ jobs:
           build_extra_args: "--compress=true"
       - uses: engineerd/setup-kind@v0.4.0
       - name: End2End
-        env:
-          DOCKER_USERNAME: ${{ github.actor }}
-          DOCKER_PASSWORD: "${{ secrets.GITHUB_TOKEN }}"
         run: |
+          docker build -t lyft/flytepropeller:${{ github.sha }} .
```
Comment (EngHabu) [inline .github/workflows/master.yml:44]: hmmm this is something I would like to dig deeper in. My understanding is that steps will run on the same machine. This means that the previous step that just build the docker image should still have the image on the docker cache. Can we just use it?

File: .github/workflows/pull_request.yml [hunk 5]
```diff
@@ -8,39 +8,18 @@ jobs:
...
-          registry: docker.pkg.github.com
+        name: "Checkout"
+
       - uses: engineerd/setup-kind@v0.4.0
       - name: End2End
-        env:
-          DOCKER_USERNAME: ${{ github.actor }}
-          DOCKER_PASSWORD: "${{ secrets.GITHUB_TOKEN }}"
         run: |
+          docker build -t lyft/flytepropeller:${{ github.sha }} .
```
Comment (EngHabu) [inline .github/workflows/pull_request.yml:16]: Can we continue to use the docker with cache action? we don't need to rebuild the image from scratch every time... specially pulling dependencies.

If my guess is correct, we should still be able to directly `kind load docker-image docker.pkg.github.com/${{PROPELLER}}` because it'll exist in the machine cache... I could be wildly wrong though :)

File: .github/workflows/master.yml (hunk 4 above)
Comment (evalsocket) [reply .github/workflows/master.yml:44]: yes i used it before also and for 100% confirmation i can add imagePullPolicy to `Never`

Comment (EngHabu) [reply .github/workflows/master.yml:44]: My point is, can we get rid of this line? because the previous step will ensure this image does exist. right?

File: Makefile [hunk 6]
```diff
@@ -47,3 +50,7 @@ golden:
 .PHONY: generate
 generate: download_tooling
 	@go generate ./...
+
+
+
+
```
Comment (EngHabu) [inline Makefile:56]: nit:
```suggestion

```

File: Makefile [hunk 7]
```diff
@@ -3,6 +3,9 @@ include boilerplate/lyft/docker_build/Makefile
 include boilerplate/lyft/golang_test_targets/Makefile
 include boilerplate/lyft/end2end/Makefile
 
+PWD := $(shell pwd)
+GIT_HASH := $(shell git log -1 --pretty=format:"%H")
+
```
Comment (EngHabu) [inline Makefile:8]: Needed?

File: .github/workflows/pull_request.yml (hunk 5 above)
Comment (EngHabu) [reply .github/workflows/pull_request.yml:16]: Is this still needed?

File: .github/workflows/pull_request.yml [hunk 8]
```diff
@@ -8,39 +8,18 @@ jobs:
...
-      - name: Push Docker Image to Github Registry
-        uses: whoan/docker-build-with-cache-action@v5
-        with:
-          username: "${{ github.actor }}"
-          password: "${{ secrets.GITHUB_TOKEN }}"
-          image_name: ${{ secrets.flytegithub_repo }}/flytepropeller
-          image_tag: ${{ github.sha }}
-          push_git_tag: true
-          registry: docker.pkg.github.com
+        name: "Checkout"
```
Comment (EngHabu) [inline .github/workflows/pull_request.yml:11]: Why delete the docker build action? can we continue to use it and just not push the image?

**Comment by evalsocket** [comment]
@EngHabu image is not availbale in secound steps that's why it's failing

File: .github/workflows/pull_request.yml [hunk 9]
```diff
@@ -8,39 +8,27 @@ jobs:
...
       - name: Push Docker Image to Github Registry
         uses: whoan/docker-build-with-cache-action@v5
         with:
           username: "${{ github.actor }}"
           password: "${{ secrets.GITHUB_TOKEN }}"
-          image_name: ${{ secrets.flytegithub_repo }}/flytepropeller
-          image_tag: ${{ github.sha }}
-          push_git_tag: true
+          image_name: lyft/flytepropeller
```
Comment (EngHabu) [inline .github/workflows/pull_request.yml:17]: If you would like to get rid of this action, I would say let's use the github-provided [cache action](https://docs.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows).

so the action would be something like:
```
- uses: github/cache
   key: "latest"
   path: "/docker_cache"
```

and in the script below (end2end or otherwise), we should follow [this](https://runnable.com/blog/distributing-docker-cache-across-hosts):
```
docker load /docker_cache/latest.tar
docker build...
docker save lyft/flytepropeller:<sha> $(docker history -q lyft/flytepropeller) > /docker_cache/latest.tar
```

If you see some other way to leverage caching please elaborate on how will this work?

**Review by EngHabu — APPROVED** [review]
This is awesome! Thank you for going through all the changes. I believe this is exactly what we need. Have you verified the cache is working? Can you make a dummy commit (you can revert it later) to see the cache being hit and used?

File: .github/workflows/master.yml [hunk 10]
```diff
@@ -28,27 +28,24 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: '0'
-      - name: Push Docker Image to Github Registry
-        uses: whoan/docker-build-with-cache-action@v5
+      - id: cache-docker
```
Comment (EngHabu) [inline .github/workflows/master.yml:31]: Any chance we can also push the image to github registry in the master workflow only? They have finally added support for anonymous pull (which was why we had to stick with dockerhub)...

https://docs.github.com/en/packages/getting-started-with-github-container-registry/about-github-container-registry

I'm ok if you want this to go in a separate PR..

Comment (evalsocket) [reply .github/workflows/master.yml:31]: no, we don't need that because in the end everyone is using dockerhub registry to pull

**Comment by evalsocket** [comment]
@EngHabu cache is working.

File: .github/workflows/master.yml (hunk 10 above)
Comment (evalsocket) [reply .github/workflows/master.yml:31]: Let's do it in seprate pr

**Comment by EngHabu** [comment]
Looking at the output [here](https://github.com/lyft/flytepropeller/runs/1156137046), I see that it's still building the image from scratch...

**Review by EngHabu — APPROVED** [review]
